src/project/tasks.js

```
@@ -158,22 +170,24 @@ export default class ProjectTaskManager {
   async runTask(task) {
     this._autoCloseSerialMonitor(task);
 
-    // Fire onWillUpload event for upload tasks and wait until all subscribers
+    // Wait for all port-owning tasks (upload*, erase*) until all subscribers
     // (e.g. ESP-Decoder) have released the serial port before starting the task.
-    if (this._isUploadTask(task)) {
+    if (this._needsPortCoordinationTask(task)) {
       try {
         await extension.fireWillUpload(this._customPort);
       } catch (err) {
         utils.notifyError('Upload Port Coordination', err);
```

ESLint

[error] 'utils' is not defined.

(no-undef)